src/repo.rs

```diff
@@ -77,13 +79,33 @@ pub fn repo_sync(repo: &str, force: bool) -> Result<()> {
 
     loop {
         println!("Running repo sync...");
-        let cmd = Command::new("repo")
+        let repo_sync = format!("repo sync -j{}", &num_cpus::get());
+        // `script` is a Unix command that takes a copy of all output to the terminal
+        // and writes it to `typescript` file.
+        // Below, explanation of `script` options.
+        // -q  Be quiet (do not write start and done messages to standard output).
+        // -e Return the exit status of the child process.
+        // -f Flush output after each write.
+        // -c Run the command rather than an interactive shell. This makes it easy for a
+        // script to capture the output of a program that behaves differently when its
+        // stdout is not a tty.
+        let mut cmd = Command::new("script")
             .current_dir(repo)
             .stdout(Stdio::piped())
             .stderr(Stdio::piped())
-            .args(["sync", "-j", &num_cpus::get().to_string()])
+            .args(["-qefc", &repo_sync])
             .spawn()
             .context("Failed to execute repo sync")?;
+
+        let stdout = cmd.stdout.take().unwrap();
+        let reader = BufReader::new(stdout);
+        let split_iter = reader
+            .split(b'\r')
+            .map(|l| String::from_utf8_lossy(&l.unwrap()).to_string());
+        for a_line in split_iter {
+            print!("{}\r", a_line);
```

Comment:
I wonder if we should actually replace \r with \n or not?

For user experience I think \r is better but for logs \n is better.

@hikalium what do you think of actually taking a lesson from our sub scirpts and deciding to do one or the other if we are a tty or not?

If we're a tty then \r, if not \n?

Reply:
I prefer not changing it implicitly, so how about just going with \n for now...?